core/perf_test/PerfTest_ViewCopy.hpp

```diff
@@ -42,220 +42,134 @@
 //@HEADER
 */
 
+#ifndef KOKKOS_CORE_PERFTEST_BENCHMARK_VIEW_COPY_HPP
+#define KOKKOS_CORE_PERFTEST_BENCHMARK_VIEW_COPY_HPP
+
 #include <Kokkos_Core.hpp>
-#include <gtest/gtest.h>
-#include <cstdio>
-#include <PerfTest_Category.hpp>
+#include <benchmark/benchmark.h>
+#include <cmath>
 
 namespace Test {
 
+void report_results(benchmark::State& state, std::size_t num_elems,
+                    double time);
+
 template <class ViewTypeA, class ViewTypeB>
-double deepcopy_view(ViewTypeA& a, ViewTypeB& b, int repeat) {
-  Kokkos::Timer timer;
-  for (int i = 0; i < repeat; i++) {
+void deepcopy_view(ViewTypeA& a, ViewTypeB& b, benchmark::State& state) {
+  for (auto _ : state) {
+    Kokkos::Timer timer;
     Kokkos::deep_copy(a, b);
+    report_results(state, a.size(), timer.seconds());
   }
-  Kokkos::fence();
-  return timer.seconds();
 }
 
 template <class LayoutA, class LayoutB>
-void run_deepcopyview_tests123(int N, int R) {
-  const int N1 = N;
+static void ViewDeepCopy_Rank1(benchmark::State& state) {
+  const int N8 = std::pow(state.range(0), 8);
```

Comment:
Any particular reason why you elected to pass `state` rather than `N`?

Reply:
This is mandatory signature (`static void Some_Benchmark(benchmark::State& state) {...}` that is expected by the `BENCHMARK` macro that registers the benchmark later.

Reply:
Oh right